File: src/main/java/rest/ProblematicaResource.java
package rest;

import entity.Invitacion;
import org.hibernate.validator.constraints.NotEmpty;
import usecase.ProblematicaUseCase;
import usecase.InvitacionUseCase;

import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;
import java.util.List;

@Path("/problematicas")
@Consumes(MediaType.APPLICATION_JSON)
@Produces(MediaType.APPLICATION_JSON)
public class ProblematicaResource {

    private final InvitacionUseCase InvitacionUseCase;
    private final ProblematicaUseCase problematicaUseCase;

    public ProblematicaResource(InvitacionUseCase invitacionUseCase, ProblematicaUseCase problematicaUseCase){
        this.InvitacionUseCase = invitacionUseCase;
        this.problematicaUseCase = problematicaUseCase;
    }

    @Path("/{idProblematica}/personas/{emailRemitente}/invitaciones")
    @GET
    public Response darPersonasInvitadas(@PathParam("idProblematica") int idProblematica,
                                         @PathParam("emailRemitente") String emailRemitente){
        List<Invitacion> personas = InvitacionUseCase.darPersonasInvitadas(emailRemitente, idProblematica);
        return Response.ok(personas).build();
    }

    @Path("/{idProblematica}")
    @POST
    public Response avanzarFase(@PathParam("idProblematica") int idProblematica,
                                @QueryParam("avanzar") @NotEmpty Boolean avanzar){
        boolean seAvanzo = problematicaUseCase.avanzarFase(idProblematica);
        return seAvanzo ?
                Response.ok().build():
                Response.status(Response.Status.BAD_REQUEST).build();
    }


}

File: src/main/java/rest/ProblematicaResource.kt
package rest

import annotation.filter.ServerEventBroadcaster
import usecase.InvitacionUseCase
import usecase.ProblematicaUseCase
import javax.validation.constraints.NotNull
import javax.ws.rs.*
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

@Path("/problematicas")
@Consumes(MediaType.APPLICATION_JSON)
@Produces(MediaType.APPLICATION_JSON)
class ProblematicaResource(private val InvitacionUseCase: InvitacionUseCase, private val problematicaUseCase: ProblematicaUseCase) {

    @Path("/{idProblematica}/personas/{emailRemitente}/invitaciones")
    @GET
    fun darPersonasInvitadas(@PathParam("idProblematica") idProblematica: Int,
                             @PathParam("emailRemitente") emailRemitente: String): Response {
        val personas = InvitacionUseCase.darPersonasInvitadas(emailRemitente, idProblematica)
        return Response.ok(personas).build()
    }

    @Path("/{idProblematica}")
    @GET
    fun darEstadoProblematica(@PathParam("idProblematica") idProblematica: Int,
                              @QueryParam("estado") @NotNull avanzar: Boolean?): Response{
        val resultado = problematicaUseCase.darEstadoProblematica(idProblematica)
        return when(resultado){
            is Error -> Response.status(Response.Status.BAD_REQUEST).entity(resultado).build()
            else -> Response.ok(resultado).build()
        }
    }

    @Path("/{idProblematica}")
    @PUT
    @ServerEventBroadcaster
    fun avanzarFase(@PathParam("idProblematica") idProblematica: Int,
                    @QueryParam("avanzar") @NotNull avanzar: Boolean?): Response {

        val resultado = problematicaUseCase.avanzarFase(idProblematica)
        return when(resultado){
            is Error -> Response.status(Response.Status.BAD_REQUEST).entity(resultado).build()
            else -> Response.ok(resultado).build()
        }
    }
}